src/Utilities/TypeTraits/GetFundamentalType.hpp

```diff
@@ -7,8 +7,15 @@
 
 #include "Utilities/NoSuchType.hpp"
 #include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/CreateHasTypeAlias.hpp"
 
 namespace tt {
+namespace detail {
+CREATE_HAS_TYPE_ALIAS(ElementType)
+CREATE_HAS_TYPE_ALIAS_V(ElementType)
```

Comment:
[optional] One idea to try is `[[maybe_unused]]` on this line. If that fails to build, we could mark the ones in the macro as that? I'm also fine ignoring and have no strong preference.

Reply:
Added a nolint block to silence this warning (new feature since clang-format 14). There's still another warning about an infinite recursion that cannot happen, which I don't think I can silence since I don't want to nolint the general code in `CachedTempBuffer`.